### PR TITLE
Lint code (based on skeleton) + Fix a weird behavior

### PR DESCRIPTION
**Pull request by cdelgehier**
- When `openio_gridinit_services` is equal to  `[]`, the global configuration is not set.
- Review the travis file for add a lint (and lint the files in this role)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,23 +8,34 @@ env:
       VERSION: 7
     - DISTRIBUTION: ubuntu
       VERSION: 16.04
+#    - DISTRIBUTION: debian
```
Comment (vincent-legoll) [inline .travis.yml:11]: Either it's supported and you add those lines uncommented or it's not and there's no need for them.

File: .travis.yml [hunk 2]
```diff
@@ -8,23 +8,34 @@ env:
...
+  # Fetch the branch with test code
   - git fetch origin docker-tests
   - git worktree add docker-tests origin/docker-tests
+  # Lint
+  - sudo pip install --upgrade pip
+  - sudo pip install ansible-lint yamllint
 
 script:
+  # Lint
+  - ansible-lint . -x ANSIBLE0016
```
Comment (vincent-legoll) [inline .travis.yml:32]: What is this ANSIBLE0016, I suppose it's an annoying warning you disable, but could you add a comment explaining which warning it is silencing ?

File: .travis.yml [hunk 3]
```diff
@@ -8,23 +8,34 @@ env:
...
+  - yamllint .
+
   # Create container and apply test playbook
   - ./docker-tests/docker-tests.sh
 
   # Run functional tests on the container
   - SUT_ID=$(docker ps -qa) ./docker-tests/functional-tests.sh
+  #- SUT_IP=172.17.0.2 ./docker-tests/functional-tests.sh
```
Comment (vincent-legoll) [inline .travis.yml:40]: What's that comment purpose ?

File: meta/main.yml [hunk 4]
```diff
@@ -1,215 +1,22 @@
+---
 galaxy_info:
-  author: Romain Acciari
+  author: Romain ACCIARI <romain@openio.io>
```
Comment (vincent-legoll) [inline meta/main.yml]: "romain.acciari@openio.io" ?

File: tasks/main.yml [hunk 5]
```diff
@@ -51,8 +51,6 @@
   template:
     src: gridinit.conf.j2
     dest: "{{ openio_gridinit_config_file }}"
-  with_items: "{{ openio_gridinit_services }}"
```
Comment (vincent-legoll) [inline tasks/main.yml:54]: The commit message does not explain that, I'd put:
`Always install the global gridinit config file, even when there's no 'present' service to setup`
Or something like that, as this also explains the purpose of the the modification

File: .travis.yml (hunk 2 above)
Comment (cdelgehier) [reply .travis.yml:32]: it fails when we restart a service on a registered_var.changed
I'll tilt, if i see again a 'flush_handlers' in the roles

File: .travis.yml (hunk 1 above)
Comment (cdelgehier) [reply .travis.yml:11]: a day, maybe
At least, travis files are equals

**Pull request merged**